lib/src/browser.dart

```diff
@@ -43,26 +42,17 @@ class Browser {
   final Function _matchesNavigator;
   final Function _parseVersion;
 
-  Version _version;
+  Version? _version;
 
-  Version get version {
-    if (_version == null) {
-      if (_parseVersion != null) {
-        _version = _parseVersion(Browser.navigator);
-      } else {
-        _version = Version(0, 0, 0);
-      }
-    }
-
-    return _version;
-  }
+  Version get version =>
+      _version ??= _parseVersion(Browser.navigator ?? TestNavigator());
```

Comment:
making `_parseVersion` and `_matchesNavigator` non nullable cleans this up a good bit